server/src/main/java/org/elasticsearch/index/mapper/PassThroughObjectMapper.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.common.Explicit;
+import org.elasticsearch.index.IndexVersion;
+import org.elasticsearch.index.mapper.MapperService.MergeReason;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Map;
+
+import static org.elasticsearch.common.xcontent.support.XContentMapValues.nodeBooleanValue;
+
+/**
+ * Mapper for pass-through objects.
+ *
+ * Pass-through objects allow creating fields inside them that can also be referenced directly in search queries.
+ * They also include parameters that affect how nested fields get configured. For instance, if parameter "time_series_dimension"
+ * is set, eligible subfields are marked as dimensions and keyword fields are additionally included in routing and tsid calculations.
+ */
+public class PassThroughObjectMapper extends ObjectMapper {
```

Comment:
In https://github.com/elastic/elasticsearch/pull/102936 I've introduced a `ObjectMapper#withoutMappers` method. You'll need to override it in this method.

Reply:
Done.